src/modules/dirwatcher.js

```diff
@@ -0,0 +1,140 @@
+import fs from 'fs';
+import AppEmitter from './../AppEmitter.js';
+
+export default class DirWatcher {
+  constructor() {
+    this.eventEmitter = new AppEmitter();
+  }
+
+  getEventEmitter() {
+    return this.eventEmitter;
+  }
+
+  watch (path, delay) {
+    let previousStats = [];
+
+    setInterval(() => {
+      fs.readdir(path, (err, newFiles) => {
+        let newStats = this.getStats(path, newFiles);
+
+        newStats.then(currentStats => {
+          if (err) {
+            console.error('Readdir error: ', err);
+            return;
+          }
+
+          let {deletedStats, addedStats} = this.checkFiles(previousStats, currentStats);
+
+          currentStats.sort(this.compareIno);
+
+          if (deletedStats.length !== 0) {
+            previousStats = this.deleteStats(deletedStats, previousStats);
+
+            //console.log('deleted');
+            this.eventEmitter.emit('changed');
+          }
+
+          if (addedStats.length !== 0) {
+            previousStats = this.addStats(addedStats, previousStats);
+
+            //console.log('added');
+            this.eventEmitter.emit('changed');
+          }
+
+          if (this.isModifiedStats(previousStats, currentStats)) {
+            previousStats = currentStats;
+
+            //console.log('changed');
+            this.eventEmitter.emit('changed');
+          }
+        });
+      });
+    }, delay);
+  }
+
+  getStats (path, files) {
+    let stats = files.map(file => {
+      let fullPath = path + file;
+
+      return this.getStat(fullPath);
+    });
+
+    return Promise.all(stats);
+  }
+
+  getStat (path) {
+    return new Promise((resolve, reject) => {
+      fs.stat(path, (err, stats) => {
+        if (err) {
+          return reject(err);
+        }
+
+        resolve(stats);
+      });
+    });
+  }
+
+  compareIno (objectA, objectB) {
+    return objectA.ino - objectB.ino;
+  }
+
+  clean (array) {
+    for (var i = 0; i < array.length; i++) {

```

Comment:
This method is strange, can you refactor them? And what do this method?

Reply:
I don't see way for refactoring, but I agree that method has unobvious behavior. So, I added comment for method.